### PR TITLE
fix: transpile ky

### PR DESCRIPTION
**Pull request by Atinux**
Fixes https://github.com/nuxt/nuxt.js/issues/5065#issuecomment-514360155

If you have an idea for testing @pi0 :)

### CODE REVIEW

File: lib/module.js [hunk 1]
```diff
@@ -90,6 +91,16 @@ function httpModule(_moduleOptions) {
       typeof options.proxy === 'object' ? options.proxy : {}
     ])
   }
+  // Add `ky` to build.transpile
+  this.options.build = this.options.build || {}
+  this.options.build.transpile = this.options.build.transpile || {}
+  // transpile only for non-modern build
+  // istanbul ignore if
+  if (semver.gte(this.nuxt.constructor.version, 'v2.9.0')) {
```
Comment (pi0) [inline lib/module.js]: This is not supporting edge versions like `2.9.0-26065925.ac00f7a6`.

image

Comment (pi0) [reply lib/module.js]:
```suggestion
  if (semver.gte(semver.coerce(this.nuxt.constructor.version), '2.9.0')) {
```

File: lib/module.js [hunk 2]
```diff
@@ -90,6 +91,16 @@ function httpModule(_moduleOptions) {
...
+    this.options.build.transpile.push(({ isLegacy }) => isLegacy ? 'ky' : null)
```
Comment (pi0) [inline lib/module.js]: Can be simplified to `isLegacy && 'key'`

Comment (Atinux) [reply lib/module.js]: Indeed, updated :)

**Comment by pi0** [comment]
@Atinux for testing, we may first wait for next edge release. Then probably ensuring that the module adds a function to transpile patterns and then validate that function behavior.

**Comment by husayt** [comment]
@pi0 @Atinux this is affecting Edge as well which is considered "modern".
So best is to transpile for modern browsers too
So instead of 
```
 this.options.build.transpile.push(({ isLegacy }) => isLegacy && 'ky')
```
just have
```
this.options.build.transpile.push('ky')
```

**Comment by pi0** [comment]
@husayt would you please make a PR?

**Pull request merged**